Promote `UseKRaft` feature gate to beta (#9518)
strimzi/strimzi-kafka-operator: documentation/modules/overview/con-configuration-points-node-pools.adoc

By using node pools, nodes can have different configuration within the same Kafka cluster.
Configuration options not specified in the node pool are inherited from the Kafka configuration.

You can deploy a Kafka cluster with one or more node pools.
The node pool configuration includes mandatory and optional settings.
Configuration for replicas, roles, and storage is mandatory.

> If you are using KRaft mode, you can specify roles for all nodes in the node pool to operate as brokers, controllers, or both.

Reviewer: ```suggestion
If you are using KRaft mode, you can specify roles that all nodes in the node pool operate as brokers, controllers, or both.
```
